### PR TITLE
Adds K8s quickstart that uses kubeadm

### PR DESCRIPTION
**Pull request by emanic**
## Description

- Changes K8s Getting Started "Overview" to "Quickstart"
- Adds new content to allow people to get up and running quickly with kubeadm
- Also changes the Jekyll version in Makefile (previous version no longer builds)

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: master/getting-started/kubernetes/index.md [hunk 1]
```diff
@@ -1,41 +1,195 @@
...
+   kube-system  etcd-jbaker-virtualbox                     1/1    Running  0         6m
+   kube-system  kube-apiserver-jbaker-virtualbox           1/1    Running  0         6m
+   kube-system  kube-controller-manager-jbaker-virtualbox  1/1    Running  0         6m
+   kube-system  kube-dns-545bc4bfd4-67qqp                  3/3    Running  0         5m
+   kube-system  kube-proxy-8fzp2                           1/1    Running  0         5m
+   kube-system  kube-scheduler-jbaker-virtualbox           1/1    Running  0         5m
+   ```
+   
+   > **Note**: You may need to give the services a little time to come
+   up. Continue issuing the `kubectl get pods --all-namespaces` command
```
Comment (2ffs2nns) [inline master/getting-started/kubernetes/index.md]: maybe add `kubectl get pods --watch --all-namespaces`

Comment (caseydavenport) [reply master/getting-started/kubernetes/index.md]: `--watch` is hard to parse as a user - it won't tell when you things are totally done.

I think `watch kubectl get pods --all-namespaces` is better.

File: master/getting-started/kubernetes/index.md [hunk 2]
```diff
@@ -1,41 +1,195 @@
...
+   kube-system  calico-node-24h85                          2/2    Running  0         2m
+   kube-system  etcd-jbaker-virtualbox                     1/1    Running  0         6m
+   kube-system  kube-apiserver-jbaker-virtualbox           1/1    Running  0         6m
+   kube-system  kube-controller-manager-jbaker-virtualbox  1/1    Running  0         6m
+   kube-system  kube-dns-545bc4bfd4-67qqp                  3/3    Running  0         5m
+   kube-system  kube-proxy-8fzp2                           1/1    Running  0         5m
+   kube-system  kube-scheduler-jbaker-virtualbox           1/1    Running  0         5m
+   ```
+   
+   > **Note**: You may need to give the services a little time to come
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/index.md]: I'd call these "pods" not services, since a service has specific meaning in k8s terminology.

File: master/getting-started/kubernetes/index.md [hunk 3]
```diff
@@ -1,41 +1,195 @@
...
+1. Execute the following commands to configure kubectl (also returned by
+   `kubeadm init`).
+
+   ```
+   mkdir -p $HOME/.kube
+   sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+   sudo chown $(id -u):$(id -g) $HOME/.kube/config
+   ```
+   
+1. Install Calico, a pod network, and a single node etcd with the
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/index.md]: I think this is misleading - Calico _is_ a pod network.

So it should probably be "Install Calico and a single node etcd" or, if you think the "pod network" bit is important, it should just say "Install Calico (a pod network) and a single node etcd"

File: master/getting-started/kubernetes/index.md [hunk 4]
```diff
@@ -1,41 +1,195 @@
...
+in approximately 15 minutes. You can use this cluster for testing and
+development.
 
-There are a number of ways to install Calico and Kubernetes.  The [installation documentation](installation)
-includes links to a number of popular guides and installers which use Calico. It also
-includes information on installing Calico on a from-scratch Kubernetes cluster using either a self-hosted Kubernetes manifest,
-or by integrating Calico into your own configuration management scripts.
+To deploy a cluster suitable for production, refer to [Installation](/getting-started/kubernetes/installation/).
 
-## Using Calico with Kubernetes
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/index.md]: So, we're replacing some content here.

ISTR that we made this page the way it was because we were directing some external links to this page, and wanted a page that helped direct users to next steps they could execute when they had a cluster.

I think we need to keep that function somehow - probably by keeping the content that was here and making the quickstart its own document. Or, we need to track down the other place that link to this and link them somewhere else instead.

File: master/getting-started/kubernetes/index.md [hunk 5]
```diff
@@ -1,41 +1,195 @@
...
 
-Once you have a Kubernetes cluster with Calico installed, the following articles will help you
-get familiar with Calico and make the most of the features that Calico provides.
+### Requirements
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/index.md]: Might want to word this as "This guide is known to work on systems which meet the following requirements" - I don't know we can / want to say it will ONLY work for these. Maybe not though, just a thought.

File: master/getting-started/kubernetes/index.md (hunk 4 above)
Comment (emanic) [reply master/getting-started/kubernetes/index.md]: I am not sure what the external documents are that point here....perhaps Mike is pointing here from his stuff? If so, it was in discussions with Mike that we determined that we should just rename the page.....actually, he wants to direct people here. Or else, let me know how we might discover the external links.

File: master/getting-started/kubernetes/index.md [hunk 6]
```diff
@@ -1,41 +1,195 @@
...
+   kubectl taint nodes --all node-role.kubernetes.io/master-
+   ```
+
+   It should return the following.
+
+   ```
+   node "<your-hostname>" untainted
+   ```
+   
+1. Switch to a root shell.
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/index.md]: Why do we do this rather than just run the following commands as `sudo`?

File: master/getting-started/kubernetes/index.md (hunk 1 above)
Comment (emanic) [reply master/getting-started/kubernetes/index.md]: Thanks @caseydavenport ... I was having trouble with `--watch` as you mention, will try this new command that you mention.

File: master/getting-started/kubernetes/index.md (hunk 4 above)
Comment (emanic) [reply master/getting-started/kubernetes/index.md]: What I can do is rename the directory to `/quickstart/` and add a redirect so that people going to `/overview/` will go to https://docs.projectcalico.org/master/getting-started/kubernetes/installation/ instead.

File: master/getting-started/kubernetes/index.md (hunk 5 above)
Comment (emanic) [reply master/getting-started/kubernetes/index.md]: In my experience, it literally does not work (reproduceably) if they don't meet these requirements. I can say that the disk space might be a little high but definitely, the others...it literally just doesn't work without these.

File: master/getting-started/kubernetes/index.md (hunk 6 above)
Comment (emanic) [reply master/getting-started/kubernetes/index.md]: It is a great question and I can't understand it either, but that's how they had it on the k8s doc site. It is definitely awkward to have to  do this.

File: master/getting-started/kubernetes/index.md (hunk 4 above)
Comment (caseydavenport) [reply master/getting-started/kubernetes/index.md]: I don't think that's right.

We have guides in other places that say "Once you're done installing Calico, go here to see what you can do" and it should have a landing page which has next steps, not more instructions for installing Calico.

Comment (caseydavenport) [reply master/getting-started/kubernetes/index.md]: I'll see if I can remember / find what the links were.

Comment (emanic) [reply master/getting-started/kubernetes/index.md]: That would help....I guess I am a little lost on this one. Thanks!

File: master/getting-started/kubernetes/index.md (hunk 1 above)
Comment (emanic) [reply master/getting-started/kubernetes/index.md]: Yes, it works great, so cool, thanks @caseydavenport

Comment (emanic) [reply master/getting-started/kubernetes/index.md]: I just can't figure out how to exit....

Comment (caseydavenport) [reply master/getting-started/kubernetes/index.md]: @emanic  you'll want "ctrl + c" for that :)

Comment (emanic) [reply master/getting-started/kubernetes/index.md]: Doh! Argh...okay thanks @caseydavenport

File: master/getting-started/kubernetes/index.md (hunk 6 above)
Comment (emanic) [reply master/getting-started/kubernetes/index.md]: I have put it on my list of follow-up items.
